src/core/lib/event_engine/ares_resolver.cc

```diff
@@ -0,0 +1,714 @@
+        GRPC_ARES_RESOLVER_TRACE_LOG("request: %p shutdown fd: %s", this,
+                                     fd_node->polled_fd->GetName());
+        fd_node->polled_fd->ShutdownLocked(
+            absl::CancelledError("AresResolver::Orphan"));
+        fd_node->already_shutdown = true;
+      }
+    }
+  }
+  Unref(DEBUG_LOCATION, "Orphan");
+}
+
+void AresResolver::LookupHostname(
+    absl::string_view name, absl::string_view default_port,
+    EventEngine::DNSResolver::LookupHostnameCallback callback) {
+  absl::string_view host;
+  absl::string_view port_s;
+  if (!grpc_core::SplitHostPort(name, &host, &port_s)) {
+    event_engine_->Run(
+        [callback = std::move(callback),
+         status = absl::InvalidArgumentError(absl::StrCat(
+             "Unparseable name: ", name))]() mutable { callback(status); });
+    return;
+  }
+  GPR_ASSERT(!host.empty());
+  if (port_s.empty()) {
+    if (default_port.empty()) {
+      event_engine_->Run([callback = std::move(callback),
+                          status = absl::InvalidArgumentError(absl::StrFormat(
+                              "No port in name %s or default_port argument",
+                              name.data()))]() mutable { callback(status); });
+      return;
+    }
+    port_s = default_port;
+  }
+  int port = 0;
+  if (port_s == "http") {
+    port = 80;
+  } else if (port_s == "https") {
+    port = 443;
+  } else if (!absl::SimpleAtoi(port_s, &port)) {
+    event_engine_->Run([callback = std::move(callback),
+                        status = absl::InvalidArgumentError(absl::StrCat(
+                            "Failed to parse port in name: ",
+                            name))]() mutable { callback(status); });
+    return;
+  }
+  // TODO(yijiem): Change this when refactoring code in
+  // src/core/lib/address_utils to use EventEngine::ResolvedAddress.
+  grpc_resolved_address addr;
+  const std::string hostport = grpc_core::JoinHostPort(host, port);
+  if (grpc_parse_ipv4_hostport(hostport.c_str(), &addr,
+                               false /* log errors */) ||
+      grpc_parse_ipv6_hostport(hostport.c_str(), &addr,
+                               false /* log errors */)) {
+    // Early out if the target is an ipv4 or ipv6 literal.
+    std::vector<EventEngine::ResolvedAddress> result;
+    result.emplace_back(reinterpret_cast<sockaddr*>(addr.addr), addr.len);
+    event_engine_->Run(
+        [callback = std::move(callback), result = std::move(result)]() mutable {
+          callback(std::move(result));
+        });
+    return;
+  }
+  grpc_core::MutexLock lock(&mutex_);
+  callback_map_.emplace(++id_, std::move(callback));
+  auto* resolver_arg = new HostnameQueryArg(this, id_, name, port);
+  if (IsIpv6LoopbackAvailable()) {
+    ares_gethostbyname(channel_, std::string(host).c_str(), AF_UNSPEC,
+                       &AresResolver::OnHostbynameDoneLocked,
+                       static_cast<void*>(resolver_arg));
```

Comment:
I don't think you need to explicitly cast to `void*` -- a function that takes a `void*` parameter should accept a pointer of any type.

Same thing throughout.

Reply:
Done.